nucliadb_node/src/bin/writer.rs

```diff
@@ -80,28 +83,33 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .insert_to_initial_state(LOAD_SCORE_KEY, node_metadata.load_score())
         .insert_to_initial_state(SHARD_COUNT_KEY, node_metadata.shard_count())
         .build()?;
-
     let node = node.start().await?;
-
-    let writer_task = tokio::spawn(start_grpc_service(grpc_driver));
-
-    let telemetry_handle = nucliadb_telemetry::sync::start_telemetry_loop();
-
     let cluster_watcher = node.cluster_watcher().await;
-    let monitor_task = tokio::task::spawn(monitor_cluster(cluster_watcher));
+    let update_handle = node.clone();
 
-    let update_task = tokio::spawn(watch_node_update(node, receiver, node_metadata));
+    nucliadb_telemetry::sync::start_telemetry_loop();
+    tokio::spawn(start_grpc_service(grpc_driver));
+    tokio::spawn(monitor_cluster(cluster_watcher));
+    tokio::spawn(watch_node_update(update_handle, receiver, node_metadata));
 
     info!("Bootstrap complete in: {:?}", start_bootstrap.elapsed());
     eprintln!("Running");
 
-    tokio::try_join!(writer_task, monitor_task, update_task)?;
-    telemetry_handle.terminate_telemetry().await;
```

Comment:
Is this doing cleanup for us? Should we still make sure all outstanding metrics/spans are flushed before exiting?

Reply:
I do not see anything on the `try_join` documentation that suggest a cleanup in case of error in one of the tasks. Right now that line was just waiting forever. 
The tasks spawned are terminated when their runtime is dropped. Maybe adding a mechanism for ensuring the other tasks dont finish abruptly should be a matter of other PR?